### PR TITLE
feat (ui): hook

### PR DESCRIPTION
**Pull request by sguiheux**

### CODE REVIEW

File: ui/src/app/model/workflow.model.ts [hunk 1]
```diff
@@ -124,9 +146,30 @@ export class WorkflowNode {
...
+            for (let i = 0; i < n.hooks.length; i++) {
+                if (n.hooks[i].id === id) {
+                    return n.hooks[i];
+                }
+            }
+            if (n.triggers) {
+                n.triggers.forEach(t => {
+                    let h = WorkflowNode.findHook(t.workflow_dest_node, id);
+                    if (h) {
+                        return h;
```
Comment (bnjjj) [inline ui/src/app/model/workflow.model.ts]: Doesn't work inside forEach

File: ui/src/app/shared/workflow/node/hook/form/node.hook.component.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Input() loading: boolean;
+    @Input('hook')
+    set hook(data: WorkflowNodeHook) {
+        if (data) {
+            this._hook = cloneDeep(data);
+            if (this.hooksModel) {
+                this.selectedHookModel = this.hooksModel.find(hm => hm.id === this._hook.model.id);
+            }
+            this.selectedHookModel = this._hook.model;
```
Comment (bnjjj) [inline ui/src/app/shared/workflow/node/hook/form/node.hook.component.ts]: missing else ?

File: ui/src/assets/i18n/fr.json [hunk 3]
```diff
@@ -537,7 +537,11 @@
   "workflow_node_delete_alert" : "ATTENTION, cela va supprimer le pipeline et tout ce qu'il déclenche",
   "workflow_node_delete_btn" : "Supprimer le pipeline",
   "workflow_node_delete_title" : "Suppression du pipeline '{{pip}}'",
-  "workflow_node_form_title" : "Pipeline à déclencher",
+  "workflow_node_hook_form_title" : "Pipeline à déclencher",
+  "workflow_node_hook_add" : "Edition d'un hook",
+  "workflow_node_hook_form_title" : "Type de hook",
+  "workflow_node_hook_form_config" : "Configuration du hook",
+  "workflow_node_hook_no_configuration" : "Aucun configuration n'est necessaire",
```
Comment (bnjjj) [inline ui/src/assets/i18n/fr.json]: aucune
nécessaire

File: ui/src/app/shared/workflow/node/hook/form/node.hook.component.ts (hunk 2 above)
Comment (sguiheux) [reply ui/src/app/shared/workflow/node/hook/form/node.hook.component.ts]: no but i have to remove line 26

**Pull request merged**